spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -329,70 +222,74 @@ object SparkTableUtil {
   private def listAvroPartition(
       partitionPath: Map[String, String],
       partitionUri: String,
-      conf: Configuration): Seq[SparkDataFile] = {
+      spec: PartitionSpec,
+      conf: Configuration): Seq[DataFile] = {
     val partition = new Path(partitionUri)
     val fs = partition.getFileSystem(conf)
 
     fs.listStatus(partition, HiddenPathFilter).filter(_.isFile).map { stat =>
-      SparkDataFile(
-        stat.getPath.toString,
-        partitionPath, "avro", stat.getLen,
-        stat.getBlockSize,
-        -1,
-        null,
-        null,
-        null,
-        null,
-        null)
+      val metrics = new Metrics(-1L, arrayToMap(null), arrayToMap(null), arrayToMap(null))
```

Comment:
Shouldn't row be a positive number?

Reply:
I think the metric is not intended to be used so it is set to an invalid value. We might need to read through whole file to get row count, right?

Reply:
Let's try to keep the logic as close to what we had before as possible.

Reply:
I think anything positive should do. I think keep it <= 0 may possibly affect some scan planning code to filter out this particular file. e.g see `org.apache.iceberg.expressions.InclusiveMetricsEvaluator`
@aokolnychyi , thoughts?

Reply:
Good catch, @rdsr. This is definitely a problem. Right now, the [`InclusiveMetricsEvaluator`](https://github.com/apache/incubator-iceberg/blob/master/api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java#L89) will remove files with negative or 0 row counts.

I don't think that the solution is to use a positive number here. The reason why this was required is that we want good stats for job planning. Setting this to -1 causes a correctness bug, but setting it to some other constant will introduce bad behavior when using the stats that are provided by Iceberg. I think we should either count the number of records, use a heuristic (file size / est. row size?), or remove support for importing Avro tables. I'm leaning toward counting the number of records.

We should also change the check in `InclusiveMetricsEvaluator` to check for files with 0 rows and allow files with -1 rows through to fix the correctness bug for existing tables that used this path to import Avro data.

Reply:
I think the number of records must be correct and precise as we want to answer some data queries with metadata (e.g. give me the number of records per partition). Updating our metrics evaluators to handle -1 seems reasonable to me as well.

Reply:
@rdsr, could you create follow-up issues so that we don't forget?

Reply:
+1 I'll do that!

Reply:
Thank you guys for the detail explanation!

Reply:
Created https://github.com/apache/incubator-iceberg/issues/809 to track this.